compiler/src/Language/Bond/Syntax/Types.hs

```diff
@@ -121,22 +123,30 @@ data TypeParam =
         }
     deriving (Eq, Show)
 
+data MethodType = Void | Unary Type | Streaming Type
+  deriving (Eq, Show)
+
 -- | Method of a service
 data Method =
     Function
         { methodAttributes :: [Attribute]   -- zero or more attributes
-        , methodResult :: Maybe Type        -- method result
+        , methodResult :: MethodType        -- method result
         , methodName :: String              -- method name
-        , methodInput :: Maybe Type         -- method parameter
+        , methodInput :: MethodType         -- method parameter
         }
     |
     Event
         { methodAttributes :: [Attribute]   -- zero or more attributes
         , methodName :: String              -- method name
-        , methodInput :: Maybe Type         -- method parameter
+        , methodInput :: MethodType         -- method parameter
         }
     deriving (Eq, Show)
 
+methodTypeToOldStyle :: MethodType -> Maybe Type
```

Comment:
A comment would be helpful here to state explicit that this was caused by adding stream and not 'old' as in 'old bond'.

Reply:
Take a look at what I've done in commit 5fdc4d6.